### ISSUE TITLE
Need some help to get started with Dapr and .Net 6 applications

### ISSUE THREAD

**Issue by GeofoxCoding** [dapr/dapr#5593]
Hi everyone,

I came across Dapr for .Net Microservices and Architecture recently and I would like to get familiar with it.

For that I was following the example project described in the official ebook page 23 and following:
https://dotnet.microsoft.com/en-us/download/e-book/dapr/pdf

with the difference that I led HTTPS-Support on for both applications.

First it was not able to start correctly and I got Exceptions in docker-compose log telling me that the certificate was missing.

I was able somehow to mount a developer certificate into the container and set the ASPNETCORE environment variables to target the same and the error is gone. I don't see any more errors but I still can't connect to my app in the browser 80, 443 ... ?

It doesn't tell me on which port my app is provided. What is wrong with my configuration?

PLEASE HELP :)! I really would like to move on with my example the app.

Dockerfiles of both WebAPI and BlazorWeb Client are nearly the same and should be okay:
```
FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
WORKDIR /app
EXPOSE 80
EXPOSE 443

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["DAPR.Test.API.csproj", "DAPR.Test.API/"]
RUN dotnet restore "DAPR.Test.API/DAPR.Test.API.csproj"
WORKDIR "/src/DAPR.Test.API"
COPY . .
RUN dotnet build "DAPR.Test.API.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "DAPR.Test.API.csproj" -c Release -o /app/publish /p:UseAppHost=false

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "DAPR.Test.API.dll"]
```

My docker-compose.yml:
```
version: '3.4'

services:
  dapr.test.frontend:
    image: ${DOCKER_REGISTRY-}bricksonawhale/dapr-test/frontend:0.0.1
    build:
      context: .
      dockerfile: DAPR.Test.Frontend/Dockerfile
    environment:
      - ASPNETCORE_ENVIRONMENT=Development
      - ASPNETCORE_Kestrel__Certificates__Default__Password=***
      - ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx
    ports:
      - "51000:50001"
    volumes:
      - ~/.aspnet/https:/https:ro

  dapr.test.frontend-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd", "--app-id", "Frontend", "--app-port", "80", "--app-ssl" ]
    depends_on:
      - dapr.test.frontend
    network_mode: "service:dapr.test.frontend"

  dapr.test.api:
    image: ${DOCKER_REGISTRY-}bricksonawhale/dapr-test/api:0.0.1
    build:
      context: .
      dockerfile: DAPR.Test.API/Dockerfile
    environment:
      - ASPNETCORE_ENVIRONMENT=Development
      - ASPNETCORE_Kestrel__Certificates__Default__Password=***
      - ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx
    ports:
      - "52000:50001"
    volumes:
      - ~/.aspnet/https:/https:ro

  dapr.test.api-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd", "--app-id", "API", "--app-port", "80", "--app-ssl" ]
    depends_on:
      - dapr.test.api
    network_mode: "service:dapr.test.api"
```

Certificate aspnetapp.pfx exists in mounted folder and provided password is correct.

Here is the end of my docker-compose log. As I said, I can't found any more errors but I also don't reach my BlazerApp at any url or port.

<img width="1624" alt="221201_Dapr_CockerCompose_Log" src="https://user-images.githubusercontent.com/79018547/205080806-cb1e396b-b895-4b6b-82c7-55df7c8f3205.png">

**Comment by mcandeia** [comment]
I don't see any portforward being declared for the 80/443 port in your docker-compose.yaml.

Can you try something like:

```yaml
version: '3.4'

services:
  dapr.test.frontend:
    image: ${DOCKER_REGISTRY-}bricksonawhale/dapr-test/frontend:0.0.1
    build:
      context: .
      dockerfile: DAPR.Test.Frontend/Dockerfile
    environment:
      - ASPNETCORE_ENVIRONMENT=Development
      - ASPNETCORE_Kestrel__Certificates__Default__Password=***
      - ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx
    ports:
      - "51000:50001"
    volumes:
      - ~/.aspnet/https:/https:ro

  dapr.test.frontend-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd", "--app-id", "Frontend", "--app-port", "80", "--app-ssl" ]
    depends_on:
      - dapr.test.frontend
    network_mode: "service:dapr.test.frontend"

  dapr.test.api:
    image: ${DOCKER_REGISTRY-}bricksonawhale/dapr-test/api:0.0.1
    build:
      context: .
      dockerfile: DAPR.Test.API/Dockerfile
    environment:
      - ASPNETCORE_ENVIRONMENT=Development
      - ASPNETCORE_Kestrel__Certificates__Default__Password=***
      - ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx
    ports:
      - "52000:50001"
      - "5001:80"
      - "5002:443"
    volumes:
      - ~/.aspnet/https:/https:ro

  dapr.test.api-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd", "--app-id", "API", "--app-port", "80", "--app-ssl" ]
    depends_on:
      - dapr.test.api
    network_mode: "service:dapr.test.api"
```

and try reach your server at 5001? tell me if it does work

**Comment by GeofoxCoding** [comment]
Hi @mcandeia,

hank you for your answer. It was helpful and useful. I added the port forwarding to my frontend instead of the api. I guess this is what you meant?

Beside that I recognized in the logs, that there was still an issue with my certificate and secrets. It cost me a lot of time but now I'm able to provide local secret and mount certificates.

I got my frontend working. Now I'm facing a new issue with my backend throwing a 500 on _daprClient.InvokeMethodAsync(...) from frontend. I'll create a new question for that.

Once again, thank yo a lot :)
